scene-lib/src/scenelib/annotations/io/classfile/ClassAnnotationSceneReader.java

```diff
@@ -1137,103 +1222,160 @@ void supplySubannotation(String fieldName, Annotation annotation) {
+     * @param fieldWriter a FieldWriter for writing the current field
+     */
+    FieldAnnotationSceneReader(int api, AElement aField, FieldVisitor fieldWriter) {
+      super(api, fieldWriter);
       this.aField = aField;
+      this.fieldWriter = fieldWriter;
     }
 
     @Override
     public AnnotationVisitor visitAnnotation(String descriptor, boolean visible) {
       if (trace) { System.out.printf("visitAnnotation(%s, %s) in %s (%s)%n", descriptor, visible, this, this.getClass()); }
-      return new AnnotationSceneReader(descriptor, visible, aField);
+      AnnotationVisitor annotationWriter = fieldWriter.visitAnnotation(descriptor, visible);
+      return new AnnotationSceneReader(this.api, descriptor, visible, aField, annotationWriter);
     }
 
     @Override
-    public TypeAnnotationVisitor visitTypeAnnotation(String descriptor, boolean visible, boolean inCode) {
-      if (trace) { System.out.printf("visitTypeAnnotation(%s, %s, %s); aField=%s, aField.type=%s in %s (%s)%n", descriptor, visible, inCode, aField, aField.type, this, this.getClass()); }
-      return new AnnotationSceneReader(descriptor, visible, aField.type);
+    public AnnotationVisitor visitTypeAnnotation(int typeRef, TypePath typePath, String descriptor, boolean visible) {
+      if (trace) { System.out.printf("visitTypeAnnotation(%s, %s, %s, %s); aField=%s, aField.type=%s in %s (%s)%n", typeRef, typePath, descriptor, visible, aField, aField.type, this, this.getClass()); }
+      AnnotationVisitor annotationWriter = fieldWriter.visitTypeAnnotation(typeRef, typePath, descriptor, visible);
+      return new TypeAnnotationSceneReader(this.api, descriptor, visible, aField.type, annotationWriter, typeRef, typePath, null, null, null, null);
     }
   }
 
   /**
    * Similarly to FieldAnnotationSceneReader, this is a MethodVisitor that
-   * only cares about visiting [extended]annotations.  Attributes other than
+   * only cares about visiting annotations.  Attributes other than
    * BootstrapMethods are ignored, all code is ignored, and visitEnd() has no
    * effect.  An AnnotationSceneReader
    * is returned for declaration and type AnnotationVisitors.  The
    * AnnotationSceneReaders have a reference to an AMethod that this is
    * visiting, and they will write out all the information to that
    * AMethod after visiting each annotation.
    */
-  private class MethodAnnotationSceneReader extends EmptyVisitor implements MethodVisitor {
+  private class MethodAnnotationSceneReader extends MethodVisitor {
 
-    // private final String name;
-    // private final String descriptor;
-    // private final String signature;
+    /** Method to be visited. */
     private final AElement aMethod;
 
-    public MethodAnnotationSceneReader(String name, String descriptor, String signature, AElement aMethod) {
+    /** MethodVisitor for writing the current method. */
+    private final MethodVisitor methodWriter;
+
+    /** Name of the local variable being visited. */
```

Comment:
This is in `MethodAnnotationSceneReader`, so I'm not sure which local variable is being visited.  Can you clarify?  Is this null when none is being visited?  Is it relevant to formal parameters, or only to real local variables?